src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/DateFormatTest.java

```diff
@@ -0,0 +1,29 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import org.junit.jupiter.api.Test;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableMap;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class DateFormatTest {
+
+    @Test
+    void testFromCodeForValidCodeReturnsDateFormat() {
+        final ImmutableMap<String, DateFormat> codeMap = ImmutableMap.<String, DateFormat>builder()
+            .put("102", DateFormat.CCYYMMDD)
+            .put("602", DateFormat.CCYY)
+            .put("610", DateFormat.CCYYMM)
+            .build();
```

Comment:
Can be simplified now
```suggestion
        final Map.of(
            "102", DateFormat.CCYYMMDD,
            "602", DateFormat.CCYY,
            "610", DateFormat.CCYYMM);
```